### PR TITLE
Poll update and delete

### PR DESCRIPTION
**Pull request by vivekpathak**

### CODE REVIEW

File: src/main/scala/com/newlibertie/pollster/impl/PollParameters.scala [hunk 1]
```diff
@@ -23,4 +23,18 @@ case class PollParameters
   if(creation_ts == None) {
     creation_ts = Some(new Date())
   }
+  override def toString: String = {
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/impl/PollParameters.scala]: can we use a more specific method of a more specific name to do the transformation we are doing in toString

File: src/test/scala/com/newlibertie/pollster/api/v1/PollApiSpec.scala [hunk 2]
```diff
@@ -46,14 +45,13 @@ class PollApiSpec extends WordSpec with Matchers with ScalatestRouteTest with La
       """.stripMargin
 
       val poll = Poll(pollDefinitionStr)
-      val pollId: String = Poll.write(poll) match {
+      val pollId: String = poll.create() match {
         case None => ""//Or handle the lack of a value another way: throw an error, etc.
```
Comment (vivekpathak) [inline src/test/scala/com/newlibertie/pollster/api/v1/PollApiSpec.scala]: please add comments to the main methods and main algorithm blocks indicating what is being done

File: src/main/scala/com/newlibertie/pollster/impl/Poll.scala [hunk 3]
```diff
@@ -28,16 +31,49 @@ object Poll {
...
+    val cur_ts = new Date()
+    (p.opening_ts before cur_ts) && (p.closing_ts after cur_ts)
+  }
+  def canOpen():Boolean = {
+    p.opening_ts after new Date()
+  }
+  def deletePoll(): Int = {
+    DataAdapter.deletePoll(p.id.get)
+  }
+  override def toString: String = {
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/impl/Poll.scala]: looks like you are buidling a specific kind of serialization.  if that is so can we name it more specifically please

File: src/main/scala/com/newlibertie/pollster/impl/Poll.scala [hunk 4]
```diff
@@ -28,16 +31,49 @@ object Poll {
...
-
-  def write(poll: Poll) = {
-    DataAdapter.createPoll(poll)
-  }
-
   def read(id:String) = {
-    DataAdapter.getPoll(id)
-
+    DataAdapter.getPoll(id) match {
+      case m:mutable.Map[String, Any] =>
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/impl/Poll.scala]: what happens when data is bad and some exception is throws.  can we emit a statement there and ensure flow is correct - even to the extent of checking that behavior in test case

File: src/main/scala/com/newlibertie/pollster/impl/Poll.scala [hunk 5]
```diff
@@ -1,25 +1,28 @@
...
     val pollJsonMap = jValue.values.asInstanceOf[Map[String, String]]
     val cryptographicParameters = if(  // string did not prescribe existing crypto params, then create new
       !pollJsonMap.contains("p") ||
-      !pollJsonMap.contains("g") ||
-      !pollJsonMap.contains("s")){
-        new CryptographicParameters()
-      }
+        !pollJsonMap.contains("g") ||
+        !pollJsonMap.contains("s")){
+      new CryptographicParameters()
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/impl/Poll.scala]: I know I had implemented the above apply.  But now it feels that two paths within the function are unsafe to some extent

Can you please add a log statement so we will always know which branch was taken

File: src/main/scala/com/newlibertie/pollster/impl/CryptographicParameters.scala [hunk 6]
```diff
@@ -23,6 +23,12 @@ case class CryptographicParameters
 )
 {
   val public_key_h = generator_g.modPow(private_key_s, large_prime_p)
+
+  override def toString: String = {
+    s"""
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/impl/CryptographicParameters.scala]: same this is not a general to string (see similar comments below)

Comment (vivekpathak) [reply src/main/scala/com/newlibertie/pollster/impl/CryptographicParameters.scala]: overwrite to string only within spirit of its original meaning : "to display data to humans"

File: src/main/scala/com/newlibertie/pollster/DataAdapter.scala [hunk 7]
```diff
@@ -82,17 +85,30 @@ object DataAdapter extends LazyLogging {
...
\ No newline at end of file
+  def updatePoll(poll:Poll): Any = {
+    val query =
+      s"""
+         |UPDATE nldb.polls
+         |SET
+         |   title = '${poll.p.title}',
+         |   tags = '${write(poll.p.tags)}',
+         |   opening_ts = '${poll.p.opening_ts.toInstant.toString.replace('T', ' ').dropRight(1)}',
+         |   closing_ts = '${poll.p.closing_ts.toInstant.toString.replace('T', ' ').dropRight(1)}',
```
Comment (vivekpathak) [inline src/main/scala/com/newlibertie/pollster/DataAdapter.scala]: you are losing accuracy of date.  please fix

File: src/main/scala/com/newlibertie/pollster/impl/Poll.scala (hunk 3 above)
Comment (usaygij) [reply src/main/scala/com/newlibertie/pollster/impl/Poll.scala]: How about using toJsonString? Else can you suggest a name as it is used to return Poll object to the REST client?

File: src/main/scala/com/newlibertie/pollster/impl/PollParameters.scala (hunk 1 above)
Comment (usaygij) [reply src/main/scala/com/newlibertie/pollster/impl/PollParameters.scala]: Same comment as above: How about using toJsonString? Else can you suggest a name as it is used to return Poll object to the REST client?

**Pull request merged**